hushline/crypto/secrets_manager.py

```diff
@@ -0,0 +1,113 @@
+"""
+A class definition to simplify the careful handling of cryptographic secrets.
+"""
+
+import warnings
+from base64 import b64decode, urlsafe_b64encode
+from hashlib import shake_256
+from pathlib import Path
+from secrets import token_bytes
+
+from aiootp.generics.canon import canonical_pack, fullblock_ljust
+from cryptography.fernet import Fernet, InvalidToken
+from passlib.hash import argon2
+
+__all__ = ["InvalidToken", "SecretsManager"]
+
+
+def truncated_b64decode(value: bytearray) -> bytearray:
+    if bool(remainder := len(value) % 4):
+        value.extend((4 - remainder) * b"=")
+    return bytearray(b64decode(value))
+
+
+class SecretsManager:
+    __slots__ = ("_secret_salt_filename", "_secret_salt_length", "_memory_cost", "_kdf")
+
+    _APP_SECRETS_DIRECTORY: Path = Path("hushline/crypto/_app_secrets")
```

Comment:
That's a good idea. Is ``/var/run/...`` available on all platforms? I'm not certain, but perhaps a private hushline home folder directory is more cross-platform? 

i.e something like:
```python
Path("~/.hushline/_app_secrets").expanduser()
```

Reply:
`/var/run/` should be available on linux/unix systems (including mac os), https://www.reddit.com/r/linux/comments/cpisy/comment/c0ua3mo/

Reply:
Since saving to the local filesystem & sharing that data between instances isn't universally available, I'll be ripping out the filesystem logic. I'll pivot towards a database approach. Perhaps a new model like ``InfrastructureAdmin``?

Reply:
Addressed by (https://github.com/scidsg/hushline/pull/411/commits/294338bb20d60ed5a2900f842e1fb432460d5d3a)